### PR TITLE
eventHandler -> reflectable, and require annotations on all JsProxy f…

### PR DESCRIPTION
**Pull request by jakemac53**
…ields/methods which you want to expose

### CODE REVIEW

**Comment by jakemac53** [comment]
fyi, I am actually going to push this to a new `1.0.0-rc.2` branch instead of merging into the `1.0.0-rc.1` branch, since it is a breaking change.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
+#### 1.0.0-rc.2
+  * The `@eventHandler` annotation has been replaced with the more general
+    `@reflectable` annotation.
+  * All fields/methods on `JsProxy` objects which you want to expose to JS need
+    to be annotated with `@reflectable`. This results in dramatic code size
+    improvement (from ~1MB to ~200KB for TodoMVC) at relatively minimal cost
```
Comment (sigmundch) [inline CHANGELOG.md]: I'd say from 400 to 200, since the other 600 are technically a bug that could be fixed without it

**Comment by sigmundch** [comment]
otherwise lgtm!

File: CHANGELOG.md (hunk 1 above)
Comment (jakemac53) [inline CHANGELOG.md]: I just removed the note entirely

**Comment by jakemac53** [comment]
Just pushed to 1.0.0-rc.2

**Pull request closed without merging**